src/extension/app-schema/app-schema-solr-test/src/test/java/org/geoserver/test/onlineTest/ComplexSolrTest.java

```diff
@@ -0,0 +1,611 @@
+            throw new RuntimeException("Invalid number of namespaces provided.");
+        }
+        for (int i = 0; i < extraNamespaces.length; i += 2) {
+            namespaces.put(extraNamespaces[i], extraNamespaces[i + 1]);
+        }
+        // add namespaces to the xpath engine
+        xpathEngine.setNamespaceContext(new SimpleNamespaceContext(namespaces));
+        return xpathEngine;
+    }
+    
+    private static Properties loadFixtureProperties(File fixtureFile) {
+        Properties properties = new Properties();
+        try (InputStream input = new FileInputStream(fixtureFile)) {
+            // load properties from fixture file
+            properties.load(input);
+            return properties;
+        } catch (Exception exception) {
+            throw new RuntimeException(
+                    String.format(
+                            "Error reading fixture file '%s'.", fixtureFile.getAbsolutePath()),
+                    exception);
+        }
+    }
+    
+    /* Helper method that moves a resource to the tests temporary directory and return the resource
+    * file path.
+    */
+    private static File moveResourceToTempDir(String resourcePath, String resourceName) {
+       // create the output file
+       File outputFile = new File(ROOT_DIRECTORY.toFile(), resourceName);
+       try (InputStream input = ComplexSolrTest.class.getResourceAsStream(resourcePath);
+               OutputStream output = new FileOutputStream(outputFile)) {
+           // copy the resource content to the output file
+           IOUtils.copy(input, output);
+       } catch (Exception exception) {
+           throw new RuntimeException("Error moving resource to temporary directory.", exception);
+       }
+       return outputFile;
+    }
+    
+    protected void checkCount(XpathEngine xpathEngine, Document document, int expectedCount, String xpath) {
+        try {
+            // evaluate the xpath and compare the number of nodes found
+            assertEquals(expectedCount, xpathEngine.getMatchingNodes(xpath, document).getLength());
+        } catch (Exception exception) {
+            throw new RuntimeException("Error evaluating xpath.", exception);
+        }
+    }
+    
+    /** Helper method that reads the content of a resource to a string. */
+    private static String readResourceContent(String resourcePath) {
+        ByteArrayOutputStream output = new ByteArrayOutputStream();
+        try (InputStream input = ComplexSolrTest.class.getResourceAsStream(resourcePath)) {
+            IOUtils.copy(input, output);
+            return new String(output.toByteArray());
+        } catch (Exception exception) {
+            throw new RuntimeException(
+                    String.format("Error reading resource '%s' content.", resourcePath), exception);
+        }
+    }
+    
+    /** Station data model class */
+    @XmlRootElement
+    public static class StationData implements Serializable {
```

Comment:
This looks duplicated with what has been wrote in the GeoTools tets, duplication of code needs to be avoided.